src/mqtt.cpp

```diff
@@ -95,7 +95,7 @@ void mqttmsg_callback(MongooseString topic, MongooseString payload) {
     DBUGF("voltage:%.1f", volts);
     evse.setVoltage(volts);
   }
-  else if (topic_string == mqtt_vehicle_soc)
+  else if (topic_string == mqtt_vehicle_soc && vehicle_data_src == VEHICLE_DATA_SRC_MQTT)
```

Comment:
Note: when I install this branch + newflags UI branch together, vehicle data are not comping to mqtt because of this new check. And the UI does not send the `vehicle_data_src` setting in the post request with the mqtt config.
Hopefully I was able to send `{"vehicle_data_src":2}` to the mqtt topic `/config/set`. I'm glad to have it ;-)

Reply:
Yes I know, that's where I was, in the vehicle page. I already had my mqtt config in place, the drop-down was on mqtt mode, but the mqtt topics for the vehicle data were not updated because the src variable was not set to mqtt mode. And saving again from the UI did not help: when introspecting the Jason payload sent, I did not see any src variable sent in the payload to tell the gateway we are in mqtt mode. That is why I was saying that updating the config fixed that (it is easier for me to update the config through mqtt)

Reply:
seems it has failed with your merges. Things are working well here

Reply:
looks you have wrong UI. We hve remove ui v1 so you don't need the env variables anymore, now the gui is inside the main tree

Reply:
I have master with gui 2 and switched the submodule to the newflags branch (your Branch) for both UI and gateway.
Sorry I might have done something wrong.
These submodules switches add more complexity.
I'm also wondering if a switch and rebuild of another gui branch is detected by the firmware rebuild

Reply:
Nope you  need to rebuid the UI if there's changes